### PR TITLE
Export a copy of the model before writing its head and names

### PR DESCRIPTION
**Pull request by raimbekovm**
## summary

- `Exporter.__call__` takes its working copy of the model before the `names` and `end2end` writes instead of 180 lines after them, so the model passed to `export()` keeps its head configuration, its training objective and its class-names object; the five `end2end` branches, the nms gate and the metadata that read them are unchanged and now all see the same copy
- a yoloworld model's cached clip encoder is kept out of the copy through `deepcopy`'s memo instead of being cleared on the caller, so the export still carries no encoder, the caller keeps the one it loaded, and the 577 MiB module is never copied
- `benchmark()` no longer deep-copies the model before each export, since the exporter isolates its own module; the copy for the pytorch row stays because predict fuses in place
- one commit per change

## measured

yolo26n.pt, imgsz=64, cpu. caller = the model object passed to `export()`; artifact = backend output on a zeros input plus metadata (ncnn param/bin md5, onnx graph md5), main vs branch.

| export | caller after export, main | caller after export, branch | artifact |
| --- | --- | --- | --- |
| torchscript `end2end=False` | end2end True -> False, E2ELoss -> v8DetectionLoss | unchanged | identical, metadata end2end=False |
| onnx `end2end=False` | same | unchanged | identical |
| ncnn (disables end2end itself) | same | unchanged | identical |
| torchscript default, torchscript `nms=True` | names object replaced | unchanged | identical, nms warning fires once on both |
| yolov8s-worldv2 after `set_classes`, torchscript | names list -> dict, clip encoder released | unchanged | identical, 0.40 s main / 0.34 s branch |

`benchmark()` with ncnn then torchscript on one model object: both rows succeed and the object is unchanged after both.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary

`Exporter.__call__` now copies the model before modifying its head, names, or export settings, preserving the caller’s model while keeping exported artifacts unchanged.

### 📊 Key Changes

- Moved the exporter’s model copy before class-name normalization, end-to-end configuration, and related metadata handling.
- Excluded a YOLOWorld cached CLIP encoder from the copied model through `deepcopy` memoization instead of clearing it on the original model.
- Removed the extra deep copy from non-PyTorch rows in `benchmark()`, while retaining the copy for the PyTorch row because prediction fuses the model in place.
- Preserved the existing YOLOWorld export warning and supported-format behavior.

### 🎯 Purpose & Impact

- Models passed to `export()` retain their original head configuration, training objective, class-names object, and YOLOWorld CLIP encoder after export.
- Benchmark exports now rely on the exporter’s internal isolation, allowing multiple formats to run on the same model object without these export-time mutations.
- Exported artifacts and their associated metadata remain unchanged by this reorganization.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @raimbekovm, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This automated message confirms your contribution was received, and an Ultralytics engineer will assist with the review. To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/engine/exporter.py [hunk 1]
```diff
@@ -672,6 +672,15 @@ def __call__(self, model=None) -> str:
...
                     "IMX export only supported for detection, pose estimation, classification, and segmentation models."
                 )
+        if isinstance(model, WorldModel):
+            LOGGER.warning(
+                "YOLOWorld (original version) export is not supported to any format. "
+                "YOLOWorldv2 models (i.e. 'yolov8s-worldv2.pt') only support export to "
+                "(torchscript, onnx, openvino, engine, coreml) formats. "
+                "See https://docs.ultralytics.com/models/yolo-world for details."
+            )
+            model.clip_model = None  # openvino int8 export error: https://github.com/ultralytics/ultralytics/pull/18445
```
Comment (UltralyticsAssistant) [inline ultralytics/engine/exporter.py]: 💡 **MEDIUM**: Moving this assignment before `deepcopy` now clears the caller's cached CLIP encoder. For example, after `model.set_classes(..., cache_clip_model=True); model.export(...); model.set_classes(...)`, `get_text_pe()` rebuilds the encoder (and may re-download it), whereas the previous ordering only cleared it on the export copy. Detach `clip_model` while creating the export copy and restore it on the caller so the encoder is not deep-copied without leaking this mutation.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Reviewed the exporter copy relocation and benchmark export path. The main export isolation change is sound, but the WorldModel path now mutates caller state by discarding its cached CLIP encoder. No other genuine issues found.

<details><summary>💬 Posted 1 inline comment</summary>

- 💡 **MEDIUM** `ultralytics/engine/exporter.py:682` Moving this assignment before `deepcopy` now clears the caller's cached CLIP encoder. For example, after `model.set_classes(..., cache_clip_model=True); model.export(...); model.set_classes(...)`, `get_text_pe()` rebuilds the encoder (and may re-download it), whereas the previous ordering only cleared it on the export copy. Detach `clip_model` while creating the export copy and restore it on the caller so the encoder is not deep-copied without leaking this mutation.
</details>

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review 2

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Since review 1, the cached WorldModel CLIP encoder is now excluded via deepcopy memo without mutating the caller, resolving the prior finding. The benchmark export path is also consistent with the exporter’s internal isolation. No remaining genuine issues found. LGTM.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review 3

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Since review 2, the WorldModel CLIP encoder exclusion remains correctly isolated through deepcopy memo, and benchmark exports now rely on the exporter’s internal copy without mutating the caller. No new regressions found in the reviewed exporter and benchmark paths. LGTM.

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review 4

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Since review 3, the exporter deepcopy memo isolation and benchmark export path remain correct. No newly introduced or unresolved behavioral issues were found in the reviewed exporter and benchmark changes. LGTM.

**Review by glenn-jocher — APPROVED** [review]
LGTM at c44343c4a7017196681f03a6298e833376d3c191. The exporter now copies before its existing head/names writes and excludes the cached WorldModel CLIP encoder through deepcopy memo without clearing the caller. Removing the redundant benchmark export copy leaves one copy at the exporter owner. Independently reviewed the entire diff. A real end2end=False TorchScript export and prediction passed; the caller retained the same names object, head mode, and state-dict keys. No added API arguments or test scaffolding.

**Comment by UltralyticsAssistant** [comment]
🎉 PR [#26052](https://github.com/ultralytics/ultralytics/pull/26052) has been merged! Thank you @raimbekovm, with contributions from @glenn-jocher, for improving exporter model isolation.

> “Alone we can do so little; together we can do so much.” — Helen Keller

By copying the model before export-time changes and safely handling the cached YOLOWorld CLIP encoder, this PR preserves the caller’s model configuration while keeping exported artifacts unchanged.
